muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/workspace-materials.js

```diff
@@ -31,8 +31,27 @@
   }
   
   $(window).load(function() {
-    if (window.location.hash && (window.location.hash.indexOf('p-') > 0)) {
-      scrollToPage(window.location.hash.substring(3), false);
+    if ($(window).data('initial-page')) {
+      scrollToPage($(window).data('initial-page'), true);
+    }
+  });
+
+  $(window).on('beforeunload', function() {
+    var tocItem = $('a.active');
```

Comment:
looking for any links with "active" class is too broad, you should limit to links inside TOC